### PR TITLE
Make it possible to configure empty "vault" auth config

### PR DESCRIPTION
**Pull request by hiyosi**
Signed-off-by: Tomoya Usami <tousami@zlab.co.jp>

**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [ ] Documentation updated?

**Affected functionality**

UpstreamAuthority `vault` plugin

**Description of change**

Fix the configuration error in the "vault" plugin that could occur when using environment variables to set parameters.

**Which issue this PR fixes**

https://github.com/spiffe/spire/pull/1611#discussion_r441685188

@amartinezfayo I appreciate your review.

### CODE REVIEW

File: pkg/server/plugin/upstreamauthority/vault/vault_test.go [hunk 1]
```diff
@@ -37,62 +38,80 @@ func (vps *VaultPluginSuite) SetupTest() {
...
+		_, err = p.Configure(ctx, req)
+		vps.Require().NoError(err)
+		vps.Require().NotNil(p.vc.vaultClient.Token())
 
-	ctx := context.Background()
-	_, err = p.Configure(ctx, req)
-	vps.Require().NoError(err)
-	vps.Require().Equal(p.vc.vaultClient.Token(), "test-token")
+		for k := range c.envKeyVal {
+			os.Unsetenv(k)
```
Comment (amartinezfayo) [inline pkg/server/plugin/upstreamauthority/vault/vault_test.go:112]: nit: maybe is better to have a `defer` call right after `Setenv `call?

**Comment by azdagron** [comment]
Is there a reason to not just fail if auth methods are configured instead of just warning?

**Comment by amartinezfayo** [comment]
>Is there a reason to not just fail if auth methods are configured instead of just warning?
I think that we should at least warn, but failing may be actually a better choice. I cannot think of a case where you intentionally want to keep two auth methods in the config.

**Comment by amartinezfayo** [comment]
I'll create a PR adding a check to fail if more than one auth method is configured.

**Comment by hiyosi** [comment]
> I'll create a PR adding a check to fail if more than one auth method is configured.

Awesome! Thank you @amartinezfayo

**Pull request merged**